dandiapi/api/services/garbage_collection/asset.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from datetime import timedelta
+import json
+from typing import TYPE_CHECKING
+
+from celery.utils.log import get_task_logger
+from django.core import serializers
+from django.db import transaction
+from django.db.models import Exists, OuterRef
+from django.utils import timezone
+from more_itertools import chunked
+
+from dandiapi.api.models import (
+    Asset,
+    GarbageCollectionEvent,
+    GarbageCollectionEventRecord,
+    Version,
+)
+
+if TYPE_CHECKING:
+    from django.db.models import QuerySet
+
+logger = get_task_logger(__name__)
+
+ASSET_EXPIRATION_TIME = timedelta(days=30)
+
+
+def get_queryset() -> QuerySet[Asset]:
+    """Get the queryset of Assets that are eligible for garbage collection."""
+    return Asset.objects.alias(
+        has_version=Exists(
+            Version.objects.filter(
+                assets=OuterRef('id'),
+            ),
+        )
+    ).filter(
+        has_version=False,
+        published=False,
+        blob__isnull=False,  # only delete assets with blobs; zarrs are not supported yet
+        modified__lt=timezone.now() - ASSET_EXPIRATION_TIME,
+    )
```

Comment:
Is there any downside to doing the following instead? This achieves the same effect using a `JOIN`, which is simpler if nothing else.

```suggestion
    return Asset.objects.filter(
        versions__isnull=False,
        published=False,
        blob__isnull=False,  # only delete assets with blobs; zarrs are not supported yet
        modified__lt=timezone.now() - ASSET_EXPIRATION_TIME,
    )
```